libs/core/langchain_core/runnables/base.py

```diff
@@ -4153,7 +4153,7 @@ def __init__(
     @property
     def InputType(self) -> Any:
         """The type of the input to this Runnable."""
-        func = getattr(self, "func", None) or getattr(self, "afunc")
+        func = getattr(self, "func", None) or self.afunc
```

Comment:
self.afunc isn't guaranteed to be set

Reply:
if it is not set then `getattr` will raise an Exception. Just like referencing `self.afunc`.
It's ruff rule B009: https://docs.astral.sh/ruff/rules/get-attr-with-constant/